src/GitLabApiClient/TreesClient.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GitLabApiClient.Internal.Http;
+using GitLabApiClient.Internal.Queries;
+using GitLabApiClient.Models.Trees.Requests;
+using GitLabApiClient.Models.Trees.Responses;
+
+namespace GitLabApiClient
+{
+    public sealed class TreesClient
+    {
+        private readonly GitLabHttpFacade _httpFacade;
+        private readonly TreeQueryBuilder _treeQueryBuilder;
+
+        internal TreesClient(GitLabHttpFacade httpFacade, TreeQueryBuilder treeQueryBuilder)
+        {
+            _httpFacade = httpFacade;
+            _treeQueryBuilder = treeQueryBuilder;
+        }
+        public async Task<IList<Tree>> GetAsync(string projectId, string branch, string path, Action<TreeQueryOptions> options)
```

Comment:
If `branch` and `path` is already part of the query option then they should not be on the method body of `GetAsync`.

the API for tree says both `branch` and `path` are optional. If none are provided it will use the default branch and repository root.